projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/linked-cases/write-linked-cases-field.component.ts

```diff
@@ -0,0 +1,187 @@
+
+@Component({
+  selector: 'ccd-write-linked-cases-field',
+  templateUrl: './write-linked-cases-field.component.html'
+})
+export class WriteLinkedCasesFieldComponent extends AbstractFieldWriteComponent implements OnInit, AfterViewInit {
+
+  public caseEditForm: FormGroup;
+  public caseDetails: CaseView;
+  public linkedCasesPage: number;
+  public linkedCasesPages = LinkedCasesPages;
+  public linkedCasesEventTriggers = LinkedCasesEventTriggers;
+  public linkedCases: CaseLink[] = [];
+  public errorMessages: ErrorMessage[] = [];
+
+  constructor(
+    private readonly appConfig: AbstractAppConfig,
+    private readonly commonDataService: CommonDataService,
+    private readonly casesService: CasesService,
+    private readonly linkedCasesService: LinkedCasesService,
+    private readonly caseEditDataService: CaseEditDataService) {
+    super();
+  }
+
+  public ngOnInit(): void {
+    // Clear validation errors
+    this.caseEditDataService.clearFormValidationErrors();
+    // Get linked case reasons from ref data
+    this.getLinkedCaseReasons();
+    this.linkedCasesService.editMode = false;
+    this.caseEditDataService.caseDetails$.subscribe({
+      next: caseDetails => this.initialiseCaseDetails(caseDetails)
+    });
+    this.caseEditDataService.caseEventTriggerName$.subscribe({
+      next: name => this.linkedCasesService.isLinkedCasesEventTrigger = (name === LinkedCasesEventTriggers.LINK_CASES)
+    });
+    this.caseEditDataService.caseEditForm$.subscribe({
+      next: editForm => this.caseEditForm = editForm
+    });
+  }
+
+  public initialiseCaseDetails(caseDetails: CaseView): void {
+    if (caseDetails) {
+      this.caseDetails = caseDetails;
+      this.linkedCasesService.caseDetails = caseDetails;
+      this.linkedCasesService.caseId = caseDetails.case_id;
+      this.linkedCasesService.caseName = this.linkedCasesService.getCaseName(caseDetails);
+      this.getLinkedCases();
+    }
+  }
+
+  public ngAfterViewInit(): void {
+    let labelField = document.getElementsByClassName('govuk-heading-l');
+    if (labelField && labelField.length) {
+      labelField[0].replaceWith('')
+    }
+    labelField = document.getElementsByClassName('heading-h2');
+    if (labelField && labelField.length) {
+      labelField[0].replaceWith('')
+    }
+  }
+
+  public onLinkedCasesStateEmitted(linkedCasesState: LinkedCasesState): void {
+    this.errorMessages = [];
+    this.caseEditDataService.clearFormValidationErrors();
+
+    if (linkedCasesState.navigateToNextPage) {
+      this.linkedCasesPage = this.getNextPage(linkedCasesState);
+      this.setContinueButtonValidationErrorMessage();
+      this.proceedToNextPage();
+    } else {
+      if (linkedCasesState.errorMessages && linkedCasesState.errorMessages.length) {
+        linkedCasesState.errorMessages.forEach((errorMessage, index) => {
+          this.caseEditDataService.addFormValidationError({ id: errorMessage.fieldId, message: errorMessage.description})
```

Comment:
@johnbenjamin-hmcts - missing semicolon

Reply:
@Daniel-Lam I have updated the code now.